CONTRIBUTING.md

```diff
@@ -2,58 +2,62 @@
+### General Guidelines
 
-Note: some of our projects are using gitflow, for these you will open pull requests against the `develop` branch.
+* Have a clear well-formatted description in the pull request. This helps reviewers and later serves as documentation in the release notes.
+* Code must adhere to the [Go formatting guidelines.](https://golang.org/doc/effective_go.html).
+* All tests must be passing.
+* New unit and integration tests should be added to ensure correctness and prevent regressions where appropriate.
+* Run linting and code formatting tools, see [the README](README.md) for details.
+* All CI checks should pass.
+* Use draft mode for PRs that are still in progress.
 
-Again, if you have a patch for a critical security vulnerability, please use our [vulnerability disclosure form][vuln_url] instead of creating a PR.  We'll follow up with you on distributing the patch before we merge it.
+### Peer Review
 
-## Code Guidelines
+This is the single most important part of introducing new code to `go-algorand`.
 
-For Go code we use the [Golang guidelines defined here](https://golang.org/doc/effective_go.html).
-* Code must adhere to the official Go formatting guidelines (i.e. uses gofmt).
-* We use **gofmt** and **golangci-lint**. Also make sure to run `make sanity` and `make generate` before opening a pull request.
-* Code must be documented adhering to the official Go commentary guidelines.
+#### Concept Review
 
-For JavaScript code we use the [MDN formatting rules](https://developer.mozilla.org/en-US/docs/MDN/Contribute/Guidelines/Code_guidelines/JavaScript).
+Because code reviews are a considerable time commitment, the first step for peer review is convincing reviewers that it is worth their time. Typically this is done by keeping changes small, writing a thorough description to clearly explain the need for a given improvement, or discussing larger changes ahead of time through one of the communication channels.
 
-For Java code we use [Oracle’s standard formatting rules for Java](https://www.oracle.com/technetwork/java/codeconventions-150003.pdf).
+If reviewers are not convinced about the merits of a change, they may reject a PR instead of reviewing it. All rejections should include the rational for how that decision was reached. It is not uncommon for this to occur. Some users opt to maintain long running forks to add features which are not suitable for the upstream repo at this time.
 
-## Communication Channels
+#### Code Review
+
+Reviewers will leave feedback directly on the pull request, typically inline with the code. This is an opportunity to discuss the changes, if a PR is left open with unresolved feedback it may eventually be closed for this reason.
```

Comment:
```suggestion
Reviewers will leave feedback directly on the pull request, typically inline with the code. This is an opportunity to discuss the changes. If a PR is left open with unresolved feedback it may eventually be closed.
```